### PR TITLE
Upgrade Beacon Network to 2nd Iteration

### LINKED ISSUE #31: Tests
Write unit tests (and maybe integration tests) when a 1.0 specification is published and apps are updated to follow that.

### LINKED ISSUE #39: Create simplified version of ELIXIR Beacon Network specification
The current specification draft is complex and we are not really using all the endpoints. We should create simplified version that contains only the endpoints that are needed.

The new spec should be aligned with GA4GH Discovery #40 #41

### LINKED ISSUE #40: Align ELIXIR Beacon Network specification with Service info spec
Related to #39 

https://github.com/ga4gh-discovery/service-info

### LINKED ISSUE #41: Align ELIXIR Beacon Network Service registry with GA4GH Service registry
https://github.com/ga4gh-discovery/service-registry

Related to #39

### LINKED ISSUE #46: Simplify the Beacon Aggregator component
Currently the Aggregator component implements all the same endpoints as the Registry. The Aggregator should be simpler, and only serve as the proxy for querying multiple Beacons.

We can simplify the component by dropping the common endpoints, keeping only those that are required for the normal functioning of the Aggregator. Removing the database, serving the Service Info from a text file, and listing known Registries in another text file.

Depends on #39

### LINKED ISSUE #47: Fetch info endpoint content from Beacon and populate Network info response with that
Currently POST method in Beacon Network info endpoint expects that all information is delivered in payload. This can be changed so that the POST query only gets Beacon URL and the Network fetches the Beacon info from its /info endpoint.

### LINKED ISSUE #50: Add coveralls
Add coveralls to travis and badge to readme

#31

### LINKED ISSUE #56: Make the API key an OTP
Change registration API key behaviour to a _One Time Password_ one.
Expire the API key `Authorization` used at Registry's `POST /services` after one use.

### PR DESCRIPTION
**Pull request by teemukataja**
This PR brings Beacon Network to its second iteration. This is a major version change and the first prototype in branch `old-version-no-support` will no longer be supported.

### Notable changes
* Beacon Network now follows [GA4GH Registry](https://github.com/ga4gh-discovery/ga4gh-service-registry/) and [Service-Info](https://github.com/ga4gh-discovery/ga4gh-service-info/) specifications;
* Aggregator no longer has a database, Registries are instead listed in a file at `/config/registries.json`;
* Extra-specification endpoints: `GET /update/services` at Registry to trigger an update of all registered services, and `DELETE /cache` at Aggregator to delete the list of cached Beacons;
* Registration `Registry POST /services` is now a pull-event instead of a push: instead of providing with a full JSON info of the service, only the URL and service type are provided. The Registry then pulls the data by itself, and is able to keep the data updated without extra maintenance effort from the registree's side;
* Code context separation: `/aggregator` and `/registry` code has been separated, but they still build into the same multipurpose image that uses the `BEACON_RUN_APP=aggregator|registry` env;
* UI has been transferred to https://github.com/CSCfi/beacon-network-ui;
* Auth has been transferred to https://github.com/CSCfi/oidc-client.

### Good to know
* Documentation has been updated;
* Unit tests have been updated;
* Integration tests need to be done;
* With UI and Auth transferred to other repositories, this repository will only contain Beacon Network APIs (Registry and Aggregator).

Fixes #31 
Fixes #39 
Fixes #40 
Fixes #41 
Fixes #46 
Fixes #47 
Fixes #50 
Fixes #56

### CODE REVIEW

File: .coverage.rc [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# .coveragerc to control coverage.py
```
Comment (blankdots) [inline .coverage.rc]:
```
Coverage.py warning: Couldn't read data from '/home/stenegru/csc/beacon-network/.coverage.rc': CoverageException: Doesn't seem to be a coverage.py data file
```
renaming maybe to `.coveragerc` will help

File: aggregator/utils/utils.py [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+
+    # Query Registries for their known Beacon services, fetch only URLs
+    service_urls = set()
+    for registry in CONFIG.registries:
+        services = await http_get_service_urls(registry.get('url', ''))  # Request URLs from Registry
+        service_urls.update(services)  # Add found URLs to set (eliminate duplicates)
+
+    # Pre-process URLS
+    service_urls = [await process_url(url) for url in service_urls]
+    service_urls = await remove_self(url_self, service_urls)
```
Comment (blankdots) [inline aggregator/utils/utils.py]: seems like lines 60+61 might make use of: https://docs.python.org/3/library/itertools.html#itertools.filterfalse

File: setup.py [hunk 3]
```diff
@@ -16,17 +16,28 @@
...
+    packages=['aggregator', 'aggregator/config', 'aggregator/endpoints',
+              'aggregator/utils', 'registry', 'registry/config',
+              'registry/endpoints', 'registry/schemas',
+              'registry/utils'],
     package_data={'': ['*.json', '*.ini']},
     install_requires=[
         'aiohttp', 'asyncpg', 'aiohttp_cors', 'uvloop',
         'asyncio', 'aiocache', 'aiomcache', 'ujson',
-        'jsonschema==3.0.0a3'
+        'jsonschema==3.0.1', 'gunicorn'
```
Comment (blankdots) [inline setup.py]: in
```suggestion
        'jsonschema==3.0.2', 'gunicorn'
```

to be on the same version as `requirements.txt`

File: .coverage.rc (hunk 1 above)
Comment (teemukataja) [reply .coverage.rc]: renamed

**Pull request merged**